### PR TITLE
Docker expose additional ports

### LINKED ISSUE #1587: exposing ports for images that does not have expose syntax in their dockerfile
When the EXPOSE directive in the docker image is not used, no ports are exposed.

But there's an alternative way to expose ports using the -p or --publish option while running a container from the command line.

Perhaps cockpit should detect that the expose syntax is not used, and offer to expose additional ports while running the container?

Or give a notification that let's the user know that no ports will be exposed.

### PR DESCRIPTION
**Pull request by dperpeet**
Docker: expose additional ports, testcase
Fixes #1587

### CODE REVIEW

File: pkg/base/cockpit.css [hunk 1]
```diff
@@ -48,7 +48,7 @@ code, kbd, pre, samp {
     background: #F5F5F5;
 }
 
-/* Contatiner is glued to the navigation and sides */
+/* Container is glued to the navigation and sides */
```
Comment (stefwalter) [inline pkg/base/cockpit.css]: This sorta thing should be a separate commit.

File: pkg/shell/cockpit-docker.js [hunk 2]
```diff
@@ -684,40 +701,126 @@ PageRunImage.prototype = {
...
+            var mapping_remove = $('<button type="button" class="btn btn-default pficon-close">');
+
+            /* if internal port is editable, allow row delete, otherwise disable modification */
+            if (port_internal_editable) {
+                mapping_remove.on('click', function () {
+                    var parent = $(this).closest("form");
+                    /* don't remove last row */
+                    if (parent.parent().children().length > 1)
+                        parent.remove();
+                    else {
```
Comment (stefwalter) [inline pkg/shell/cockpit-docker.js]: If one condition has braces, I think we should make them all have braces. It's far more readable that way. WDYT?

File: pkg/shell/cockpit-docker.js [hunk 3]
```diff
@@ -684,40 +701,126 @@ PageRunImage.prototype = {
...
+                port_internal_element.prop('disabled', true);
+                protocol_select.prop('disabled', true);
+                mapping_remove.prop('disabled', true);
+            }
+
+            var portmapping = {};
+            portmapping.html =
+                    $('<form class="form-inline">')
+                    .append($('<div class="form-group">')
+                        .append(port_internal_element))
```
Comment (stefwalter) [inline pkg/shell/cockpit-docker.js]: Suggest putting closing parens on their own line, to increase readability.

File: pkg/shell/cockpit-docker.js (hunk 2 above)
Comment (dperpeet) [inline pkg/shell/cockpit-docker.js]: I tried both options when I wrote this, but readability wasn't really increased.
[Coding style](https://github.com/GNOME/gtk/blob/ba06066158cc2c4e68a8de5706349a2c221c5c58/docs/CODING-STYLE#L121-L142) says there should be braces.

File: pkg/base/cockpit.css (hunk 1 above)
Comment (dperpeet) [inline pkg/base/cockpit.css]: moved to separate commit

**Comment by stefwalter** [comment]
Using: fedora/apache:latest image. When i try to map predefined port 80 ->33333 and add an additonal port 8080 -> 80028, i get this error:

screenshot from 2015-01-22 11 45 33

**Comment by stefwalter** [comment]
> Using: fedora/apache:latest image. When i try to map predefined port 80 ->33333 and add an additonal port 8080 -> 80028, i get this error:

Port is too high. We should be highlighting the field as an error.

**Comment by dperpeet** [comment]
We could add a warning if a user enters invalid port numbers or color the field accordingly.

**Comment by dperpeet** [comment]
Input validation moved to separate issue #1708
